### PR TITLE
test: migrate integration coverage to unit tests

### PR DESCRIPTION
**Pull request by keulinho**
### 1. Why is this change necessary?

Several integration tests only exercised pure branch logic or duplicated existing unit coverage. Moving those cases into focused unit tests improves unit coverage and reduces integration-suite runtime while keeping integration coverage for DB, container, and service-wiring behavior.

### 2. What does this change do, exactly?

- Removes duplicate/pure integration cases for store middleware, Twig template loading, app flow action loading, media metadata/file fetching, date serializers, comparison extension, and line item group builder coverage.
- Adds or extends unit tests for the migrated behavior, including date/date-time serializer branches, media file fetch edge cases, metadata extraction with unit fixtures, middleware passthrough behavior, comparison operators, and line item group scenarios.
- Keeps the remaining integration tests focused on persistence, container wiring, or real service-level behavior.

### 3. Describe each step to reproduce the issue or behaviour.

This is a test-suite cleanup. Run the focused unit tests for the migrated coverage:

`php -d memory_limit=-1 vendor/bin/phpunit --no-configuration --bootstrap vendor/autoload.php tests/unit/Core/Framework/DataAbstractionLayer/FieldSerializer/DateFieldSerializerTest.php tests/unit/Core/Framework/DataAbstractionLayer/FieldSerializer/DateTimeFieldSerializerTest.php tests/unit/Core/Content/Media/File/FileFetcherTest.php tests/unit/Core/Content/Media/File/FileUrlValidatorTest.php tests/unit/Core/System/SystemConfig/SystemConfigServiceTest.php tests/unit/Core/Framework/Store/Services/StoreSessionExpiredMiddlewareTest.php tests/unit/Core/Framework/Store/Services/ShopSecretInvalidMiddlewareTest.php tests/unit/Core/Framework/Adapter/Twig/EntityTemplateLoaderTest.php tests/unit/Core/Framework/App/Flow/Action/AppFlowActionLoadedSubscriberTest.php tests/unit/Core/Content/MeasurementSystem/TwigExtension/MeasurementConvertUnitTwigFilterTest.php tests/unit/Core/Content/MailTemplate/Service/MailTemplateServiceTest.php tests/unit/Core/Checkout/Cart/LineItem/Group/LineItemGroupBuilderTest.php tests/unit/Core/Content/Media/Metadata/MetadataLoaderTest.php tests/unit/Core/Framework/Adapter/Twig/Extension/ComparisonExtensionTest.php`

### 4. Please link to the relevant issues (if any).

None.

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have updated developer-facing release notes if this change is **relevant** for external developers:
  - Add a short entry to `RELEASE_INFO-6.<major>.md` under “Upcoming” for informational changes, including the consequences of the change and how it affects external developers.
  - Add an `UPGRADE` section in `UPGRADE-6.<next-major>.md` for breaking changes (what/why/impact/how to adapt).
  - See the [Documenting a Release Process](https://github.com/shopware/shopware/blob/trunk/delivery-process/documenting-a-release.md) for details.
- [x] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfilled them

### CODE REVIEW

File: tests/integration/Core/Checkout/Cart/LineItem/Group/LineItemGroupBuilderTest.php [hunk 1]
```diff
@@ -1,474 +0,0 @@
...
-use Shopware\Tests\Unit\Core\Checkout\Cart\LineItem\Group\Helpers\Fakes\FakeTakeAllRuleMatcher;
-use Shopware\Tests\Unit\Core\Checkout\Cart\LineItem\Group\Helpers\Traits\LineItemGroupTestFixtureBehaviour;
-use Shopware\Tests\Unit\Core\Checkout\Cart\LineItem\Group\Helpers\Traits\LineItemTestFixtureBehaviour;
-use Shopware\Tests\Unit\Core\Checkout\Cart\LineItem\Group\Helpers\Traits\RulesTestFixtureBehaviour;
-
-/**
- * @internal
- */
-#[Package('checkout')]
-class LineItemGroupBuilderTest extends TestCase
```
Comment (keulinho) [inline tests/integration/Core/Checkout/Cart/LineItem/Group/LineItemGroupBuilderTest.php:52]: this might look dangerous at first, but when you look at it this test case was already a pure unit test and it mocked everything and didn't really interact with anything in the container, now all the cases are moved into the real unit test that also already existed for this class

**Review by cyl3x — APPROVED** [review]

**Review by nfortier-shopware — APPROVED** [review]
